### ISSUE TITLE
can you rebuild docker.io/elementsproject/lightningd ?

### ISSUE THREAD

**Issue by x3nb63** [ElementsProject/lightning#4795]
### Issue and Steps to Reproduce

DNS lookups fail in the container, when running on a k3s cluster. As a consequence lightningd fails to connect to bitcoind.

I do have (many) containers that use the clusters DNS just fine... so its not the cluster per se. This far I tracked the issue down to versions of Alpine where DNS succeeds or fails depending on the Alpine version chosen. This test I did with running a alpine shell as a _sidecar_ to the lightningd container - so I am right in the same pod (same host, same virtual nic, same packet filter rules that apply ...) - an I can toggle whether name lookups succeed or not.

From alpine I know there have been many updates on the resolving library in their busybox core ... I have no idea what Debian does or does not here, but given they have a newer build than current `elementsproject/lightningd:latest` points to, I think its worth doing.

also: does it need to be `buster`? Perhaps lightningd works just fine on a newer release?

**Comment by cdecker** [comment]
Sounds like something that we might want to look into. Can you try and see how resolution is configured in a non-lightningd container and a lightningd container (presumably using `kubectl exec` to run in the container itself).

Depending on how these differ we might have a better idea on how to fix this up in the docker image.

**Comment by x3nb63** [comment]
all pods have a `resolv.conf` file pointing to the kube-dns service IP. The service itself is then backed by a coredns pod. There is no difference other than the namespace in the `search ...` list.

however, the question of query with FQDN / without FQDN does not matter in my tests, DNS works in the container or it does not.

testing with `nslookup` tool sometimes reports a answer but at the same time says: `*** Can't find host.domain.tld: No answer`.

**Comment by cdecker** [comment]
It seems indeed like the kubernetes DNS resolution may have some issues with alpine containers:

 - https://stackoverflow.com/questions/65181012/does-alpine-have-known-dns-issue-within-kubernetes#answer-65593511
 - https://github.com/kubernetes-sigs/kind/issues/442
 - https://github.com/kubernetes/kubernetes/issues/30215

I'll look into how we can address that without requiring changes to the kubernetes config.

**Comment by x3nb63** [comment]
yes, I've found many issues with this and many hint to musl and/or busybox. ... that is, to the "core" of Alpine and the reason why its so small and efficient.

Yet your Dockerfile uses buster-slim ... so I am not sure this is it - unless this slimification switches to busybox and/or some other libc ...? That I can't tell.

**Comment by rustyrussell** [comment]
AI cleanup of old issues.  If wrong, please open a new issue.


Triage decision summary:
Request to rebuild Docker image with updated Alpine/Debian base to fix DNS resolution issues in Kubernetes. Likely resolved by subsequent image rebuilds or base image updates over time.
